src/genai/eyfs/eyfs.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import string
+
+from typing import Dict
+from typing import List
+
+import openai
+
+from openai.error import APIConnectionError
+from openai.error import APIError
+from openai.error import RateLimitError
+from openai.error import ServiceUnavailableError
+from openai.error import Timeout
+from tenacity import before_sleep_log
+from tenacity import retry
+from tenacity import retry_if_exception_type
+from tenacity import stop_after_attempt
+from tenacity import wait_exponential
+
+
+logger = logging.getLogger(__name__)
+
+
+class ActivityGenerator:
+    """Generate tokens using OpenAI's API.
+
+    TODO:
+    - Streaming
+    - Async calls
+
+    """
+
+    @classmethod
+    def generate(
+        cls,
+        messages: List[Dict],
+        message_kwargs: Dict,
+        temperature: float = 0.0,
+        **openai_kwargs,
+    ) -> Dict:
+        """Generate text using OpenAI's API."""
```

Comment:
I'm not sure what is `cls`? Perhaps we can explain in the docstring...

Reply:
Oh is that `self` ?

Reply:
Adding this as a reference on our chat earlier today: https://peps.python.org/pep-0008/#function-and-method-arguments